src/pages/NewTarget/NewTarget.jsx

```diff
@@ -0,0 +1,121 @@
+
+  const t = useTranslation();
+
+  const { data: topics = [] } = useGetTopicsQuery();
+
+  //const [newTarget, { isLoading, isSuccess, error }] = useNewTargetMutation();
+
+  const [addTarget, { isLoading, isSuccess, error }] = useAddTargetMutation();
+
+  const schema = z.object({
+    area: z.string().min(1, { message: t('newTarget.errors') }),
+    title: z.string().min(1, { message: t('newTarget.errors') }),
+    topic: z.string().min(1, { message: t('newTarget.errors') }),
+  });
+
+  const onSubmit = data => {
+    const target = {
+      radius: parseInt(data.area),
+      title: data.title,
+      lat: state.coordinates.lat,
+      lng: state.coordinates.lng,
+      topic_id: state.topic,
+    };
+    addTarget(target);
+  };
+
+  const {
+    register,
+    setValue,
+    handleSubmit,
+    formState: { errors },
+  } = useForm({ resolver: zodResolver(schema) });
+
+  const handleFeedback = useCallback(() => {
+    if (isLoading) {
+      setFeedback(prev => ({ ...prev, loading: true }));
+    }
+    if (isSuccess) {
+      setFeedback(prev => ({ ...prev, loading: false, success: true }));
+    }
+    if (error) {
+      setFeedback(prev => ({ ...prev, loading: false, error: true }));
+    }
+  }, [isLoading, isSuccess, error]);
+
+  useEffect(() => {
+    handleFeedback();
+  }, [handleFeedback]);
+
+  return (
+    <div className="new">
+      <img src={target} alt="target" className="new__target-icon" />
+      <p className="new__title">{t('newTarget.title')}</p>
+
+      <form className="new__form" onSubmit={handleSubmit(onSubmit)}>
+        <label htmlFor="area" className="form__label new__label">
+          {t('newTarget.labels.area')}
+        </label>
+        <Input
+          register={register}
+          error={errors.area}
+          name="area"
+          className="new__input"
+          type="number"
+        />
+        <label htmlFor="title" className="form__label new__label">
+          {t('newTarget.labels.title')}
+        </label>
+        <Input
+          register={register}
+          error={errors.title}
+          name="title"
+          className="new__input"
+          placeholder={t('newTarget.placeholder.title')}
+        />
+        <label htmlFor="topic" className="form__label new__label">
+          {t('newTarget.labels.topic')}
+        </label>
+        <div className="new__select">
+          <Select
+            register={register}
+            options={topics.topics}
+            errors={errors}
+            placeholder={t('newTarget.placeholder.topic')}
+            setValue={setValue}
+          />
+        </div>
+        <div className="new__btn-container--fullwidth">
+          <div className="new__btn-container">
+            <Button type="submit">
+              {feedback.loading ? t('newTarget.btnSaving') : t('newTarget.btn')}
+            </Button>
+          </div>
+        </div>
+      </form>
+      {error && <p className="new__error">{error.data.errors.user[0]}</p>}
```

Comment:
WDYT if you code a function that gets the errors from the API? Then you could re-use it whenever you need it. 
And you should take in count what could happen if `error.data.errors.user[0]` structure changes in the future.

Reply:
I see what you mean. I ended up replacing the error with a generic message because it doesn't always come at the same object key. Sometimes it is {target_limit: "You've reached your target limit"} and sometimes it is {user: ["you can't be at two conversations at onces']}